### DISCUSSION TITLE
Option to autocache all played songs and pre-download songs up to a select number from the play-queue

### DISCUSSION THREAD

**Discussion by jeffgt14** [CappielloAntonio/tempo#51] [Ideas]
Stealing another useful feature (for me) that DSub has. The ability to cache all songs that are played for offline use. DSub lets you select any amount of GB to use for cache (say 40GB for example) and caches all songs played up to 40GB. Once that 40GB is used up, the oldest played songs are subsequently removed to make room for the latest played songs to be cached.

It would also be great for songs from the play queue to autodownload up to a select number. Say if the play queue is 12 songs, you can select any number (say 7) that start to cache ahead of time. This helps prevent issues if going in and out of service to prevent auto from stopping (typically an issue when streaming higher res files like FLAC that take a tad bit longer to download.

**Comment by jibbel23** [comment]
I've been using the OG subsonic client for years and only feature I've been looking for is gapless playback to make a switch. I demo'd tempo and it seems OK but the lack of caching is a dealbreaker. Would also love to see this implemented.

**Comment by SteveDinn** [comment]
Totally agree. The way DSub handled caching of tracks was *chef's kiss*.

**Comment by yashaskm11** [comment]
This is the only feature holding me back from using Tempo, sticking to DSub untill this would be implemented.
